shell/platform/android/platform_view_android.cc

```diff
@@ -59,18 +54,56 @@ class PlatformMessageResponseAndroid : public blink::PlatformMessageResponse {
   ftl::WeakPtr<PlatformView> view_;
 };
 
-}  // namespace
+static std::unique_ptr<AndroidSurface> InitializePlatformSurfaceGL() {
+  const PlatformView::SurfaceConfig offscreen_config = {
+      .red_bits = 8,
+      .green_bits = 8,
+      .blue_bits = 8,
+      .alpha_bits = 8,
+      .depth_bits = 0,
+      .stencil_bits = 0,
+  };
+  auto surface = std::make_unique<AndroidSurfaceGL>(offscreen_config);
+  return surface->IsOffscreenContextValid() ? std::move(surface) : nullptr;
+}
+
+static std::unique_ptr<AndroidSurface> InitializePlatformSurfaceVulkan() {
+#if SHELL_ENABLE_VULKAN
+  auto surface = std::make_unique<AndroidSurfaceVulkan>();
+  return surface->IsValid() ? std::move(surface) : nullptr;
+#else   // SHELL_ENABLE_VULKAN
+  return nullptr;
+#endif  // SHELL_ENABLE_VULKAN
+}
+
+static std::unique_ptr<AndroidSurface> InitializePlatformSurface() {
+  if (auto surface = InitializePlatformSurfaceVulkan()) {
+    FTL_DLOG(INFO) << "Vulkan surface initialized.";
+    return surface;
+  }
+
+  FTL_DLOG(INFO)
+      << "Could not initialize Vulkan surface. Falling back to OpenGL.";
+
+  if (auto surface = InitializePlatformSurfaceGL()) {
+    FTL_DLOG(INFO) << "GL surface initialized.";
+    return surface;
+  }
+
+  FTL_CHECK(false) << "Could not initialize either the Vulkan or OpenGL "
```

Comment:
I have that in another patch that depends on the infrastructure created for Vulkan. So, not yet.